processing/src/main/java/org/apache/carbondata/processing/loading/sort/unsafe/holder/UnsafeSortTempFileChunkHolder.java

```diff
@@ -296,11 +200,12 @@ private void fillDataForPrefetch() {
    * @throws CarbonSortKeyAndGroupByException
    */
   private Object[] getRowFromStream() throws CarbonSortKeyAndGroupByException {
-    Object[] row = new Object[dimensionCount + measureCount];
+    Object[] row = new Object[sortParameters.getDimColCount()
+        + sortParameters.getComplexDimColCount() + sortParameters.getMeasureColCount()];
```

Comment:
I think there is difference with changed code compare to old code. Please double check the changes. Complex dimensions may not work

Reply:
I think better keep the old of keeping variables at class level instead of calling method for each row.

Reply:
Fine~ I also find the same issues in other files and will fix it at the same time.